### PR TITLE
[`EfficientSAM`] Add EfficientSAM to the library

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As per title, this PR adds EfficientSAM, a new architecture from https://github.com/yformer/EfficientSAM that is similar than SAM architecture but with the benefit of being much smaller. 

Draft for now 

@xenova @yformer

### CODE REVIEW

**Comment by rishabh063** [comment]
Hey when can we expect this to merge ?

**Comment by younesbelkada** [comment]
Hi @rishabh063 
Thanks very much for your interest in this PR - hopefully quite soon, I need some time to make sure logits match then we can merge

**Comment by rishabh063** [comment]
Can you share some resources with me to bring me to speed 

I will also try to contribute if i can

**Comment by rishabh063** [comment]
hey @younesbelkada any updates on this ?

**Comment by younesbelkada** [comment]
hi @rishabh063 unfortunately I had to work on more urgent things now, I will come back to this ASAP. If you have bandwith you can also take over the PR
